astroplan/constraints.py

```diff
@@ -394,7 +395,81 @@ def compute_constraint(self, times, observer, targets):
                              "MoonSeparationConstraint.")
         return mask
 
+    
+class DarkGreyBrightConstraint(Constraint):
+    """
+    Constrain whether it is dark, gray or bright time or how bright.
+    """
+    # TODO: make a new test
+    @u.quantity_input(horizon=u.deg)
+    def __init__(self, lunar_horizon=90*u.deg, min_moon_illumination=None,
+                                                max_moon_illumination=None):
+        '''
+        Parameters
+        ----------
+        Lunar_horizon : `~astropy.units.Quantity` (optional)
+            The altitude of the ``horizon`` and defaults to 90 degrees       
```

Comment:
I think this should default to 0 degrees. The way this ```Constraint``` should work is that it is satisfied if the moon illumination fractions are satisfied *or* if the moon is below the specified horizon. 

This is because this constraint will most often be used to ensure that the night sky is not too bright because of the Moon, and the night sky brightness drops dramatically once the Moon has set (see [here](https://confluence.lsstcorp.org/display/SIM/Sky+Brightness+Observations)).

The docstring should make this behaviour clear.